### ISSUE TITLE
Automatic sidecar-injector webhook broken (Istio 1.7.x) on EKS (1.17) Redux

### ISSUE THREAD

**Issue by jacogreyling** [istio/istio#27585]
**Bug description**

I know a lot of people have raised issues regarding this problem, especially in the following two GitHub issues:

- [Rolling upgrade breaks sidecar-injector webhook on EKS (failed calling admission webhook "sidecar-injector.istio.io": context deadline exceeded)](https://github.com/istio/istio/issues/13840)
- [Admission webhooks affected by dead tcp connections](https://github.com/kubernetes/kubernetes/issues/80313)

However most of this was related to either EKS clusters on earlier versions of Kubernetes, or Istio 1.6 or earlier. There seems to be two proposed solutions as outlined in the above.

(a) Update the istio-sidecar-injector deployment with:
```spec:
  template:
    spec:
      hostNetwork: true
```

(b) Update the istio-sidecar-injector deployment with:
```spec:
  template:
    spec:
      containers:
      - name: sidecar-injector-webhook
        env:
        - name: GODEBUG
          value: http2server=0
```

Both of these options expect the administrator to update the **istio-sidecar-injector** deployment. The problem is with Istio 1.7.x the architecture completely changed and there is no sidecar deployment anymore. I've tried updating the **istiod** deployment and the **istio-sidecar-injector** ConfigMap but still the same issue persists.

Question: Am I updating the resources in the right place, given the changes to the architecture is there any examples of where I should update the resources??

To give a little more context, we're running AWS EKS (1.17) with Cilium 1.8 in **overlay** mode. We can manually inject the sidecar with istioctl. But the automatic sidecar injector breaks with the following error:

```LAST SEEN   TYPE      REASON         OBJECT                              MESSAGE
14m         Warning   FailedCreate   replicaset/xxxxx-66b7b6f974   Error creating: Internal error occurred: failed calling webhook "sidecar-injector.istio.io": Post https://istiod.istio-system.svc:443/inject?timeout=30s: context deadline exceeded
31m         Warning   FailedCreate   replicaset/xxxxx-66b7b6f974   Error creating: Internal error occurred: failed calling webhook "sidecar-injector.istio.io": Post https://istiod.istio-system.svc:443/inject?timeout=30s: dial tcp x.x.x.x:15017: i/o timeout
```

To make things even more confusing, the official documentation [here](https://istio.io/latest/docs/ops/common-problems/injection/#no-such-hosts-or-no-endpoints-available-errors-in-deployment-status) talks about running the following command `kubectl -n istio-system get pod -listio=sidecar-injector` which I assume relates to the previous version of Istio?

Any thoughts?

[ ] Docs
[X] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

I expect the automatic sidecar-injector to work as per the official documentation here: https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/

**Steps to reproduce the bug**

Install EKS via eksctl (manifests can be provided on request), and follow the documentation [here](https://istio.io/latest/docs/setup/install/operator/) to install Istio via the Istio Operator steps. Then install Cilium for EKS clusters as outlined [here](https://docs.cilium.io/en/v1.8/gettingstarted/k8s-install-eks/). Please note, do NOT install it in ENI  mode but rather keep the stock standard installation (exclude the lines for global.eni=true and global.tunnel=disabled from the helm command).

Then install an example app with the given namespace set to auto sidecar injection.

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**

Istio:
Istio 1.7.1

Cilium (overlay mode):
Cilium 1.8.3

Kubernetes:
Client Version: v1.16.6-beta.0
Server Version: v1.17.9-eks-4c6976

Helm:
version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"dirty", GoVersion:"go1.14.5"}

**How was Istio installed?**

See steps above to reproduce. (Via helm, and Istio Operator)

**Environment where bug was observed (cloud vendor, OS, etc)**

See above

**Comment by jacogreyling** [comment]
UPDATE: We managed to get it working. The problem is that the EKS control plane **cannot** talk back to the services inside the cluster when in overlay network mode. This means that if you run CNI's like Cilium in overlay mode (`global.eni=false` and `global.tunnel=enabled`) it will break communications. This affects a lot more than Istio since anything that the control plane needs access to like CR's (e.g. Mutating Webhooks) will fail.

We managed to solve this by running the affected deployments with `hostNetwork: true` configured. This is of course not a long term solution but at least provides us with a way forward.

However, part of the problem still persists in that the Istio documentation is not up to date. The following document https://istio.io/latest/docs/ops/common-problems/injection/#no-such-hosts-or-no-endpoints-available-errors-in-deployment-status still talks about the istio-sidecar-injector deployment that does not exist in Istio 1.7.x anymore.
